### PR TITLE
Fix simple matrix `getType` methods

### PR DESCRIPTION
**Pull request by FlorentinD**
based on issue #139 

Now implicitly converting to the required type if needed.
Also added a matrix equals for combination of float and double matrices

### CODE REVIEW

**Comment by FlorentinD** [comment]
Forgot to run `spotlessJavaApply` again. Maybe we could add a PR template with a checklist, which includes `Run spotlessJavaApply`

**Comment by lessthanoptimal** [comment]
Might have some tweaks. On a trip at the moment but I'll finish review in a few days I think. Thanks for the quick PR!

**Comment by FlorentinD** [comment]
:wave:  a short reminder to review this PR (@lessthanoptimal)

File: main/ejml-core/src/org/ejml/ops/ConvertMatrixType.java [hunk 1]
```diff
@@ -36,12 +35,13 @@
     public static Matrix convert( Matrix matrix, MatrixType desired ) {
         Matrix m = null;
 
+        if (matrix.getType().equals(desired)) {
```
Comment (lessthanoptimal) [inline main/ejml-core/src/org/ejml/ops/ConvertMatrixType.java]: I think the original behavior here was correct, but the JavaDoc needs to be updated to make explicit what the contract is. The logic is that in every case a new instance should be created. That way you know you're not modifying the original and you don't need to check for the special case where its the same matrix type..

**Review by lessthanoptimal — CHANGES_REQUESTED** [review]
One comment and if you disagree we should discuss what the correct behavior should be. Either way the contract needs to be made explicit.

File: main/ejml-core/src/org/ejml/ops/ConvertMatrixType.java (hunk 1 above)
Comment (FlorentinD) [reply main/ejml-core/src/org/ejml/ops/ConvertMatrixType.java]: Agree, always making a copy is easier to handle for the user

**Review by lessthanoptimal — APPROVED** [review]

**Pull request merged**